edb/server/protocol/auth_ext/http.py

```diff
@@ -0,0 +1,169 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2022-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import datetime
+import http
+import json
+import urllib.parse
+import base64
+
+from typing import *
+from jwcrypto import jwk, jwt
+
+from edb import errors as edb_errors
+from edb.common import debug
+from edb.common import markup
+
+from . import errors
+from . import util
+
+from edb.server.protocol.protocol import HttpRequest, HttpResponse
+
+
+class Router:
+    def __init__(self, db: Any, base_path: str):
```

Comment:
For some reason, I was getting runtime errors when I was trying to use the imported `Database` from `dbview`...

```
        > class Client:
    /Users/scotttrinh/edgedb/edgedb/edb/server/protocol/auth_ext/oauth.py, line 29, in Client
        > def __init__(self, db: Database, provider: str):

NameError: name 'Database' is not defined
```

Reply:
I had a `dbview.pyi` file initially that I was defining the `Database` class with in detail since the Cython classes are not picked up by my tooling (seems like this is a "by design" thing based on reading a bit), which I think was throwing this off. It works at runtime if I remove the `pyi` typing stub and just import it directly, but then the type is basically `Unknown` which isn't very helpful.

Any advice? I'm not going to get stuck on this, but I'm hoping there is a way to get useful typing here.